src/utils/currency/contractPrice.ts

```diff
@@ -0,0 +1,99 @@
+import { Network, SupportedCurrencies } from '~gql';
+
+import { currencyApiConfig, coinGeckoMappings } from './config.ts';
+import {
+  type FetchCurrentPriceArgs,
+  type CoinGeckoSupportedCurrencies,
+  type SupportedChains,
+  type TokenContractPriceSuccessResponse,
+  type TokenContractPriceResponse,
+} from './types.ts';
+import { buildAPIEndpoint, fetchData, mapToAPIFormat } from './utils.ts';
+
+// The functions defined in this file assume something about the shape of the api response.
+// If that changes, or if we change the api, these functions will need to be updated.
+
+const { chains, currencies } = coinGeckoMappings;
+const buildTokenContractCoinGeckoURL = (
+  contractAddress: string,
+  chainId: SupportedChains = Network.Gnosis,
+  conversionDenomination: CoinGeckoSupportedCurrencies = SupportedCurrencies.Usd,
+) => {
+  const chain = mapToAPIFormat(chains, chainId);
+  const denomination = mapToAPIFormat(currencies, conversionDenomination);
+  const { url, searchParams } =
+    currencyApiConfig.endpoints.tokenPriceByContract;
+
+  return buildAPIEndpoint(new URL(`${url}/${chain}`), {
+    [searchParams.from]: contractAddress,
+    [searchParams.to]: denomination,
+    [searchParams.api]: process.env.COINGECKO_API_KEY ?? '',
+  });
+};
+
+const extractContractPriceFromResponse = (
+  data: TokenContractPriceSuccessResponse,
+  contractAddress: string,
+  conversionDenomination: CoinGeckoSupportedCurrencies,
+) => {
+  try {
+    return data[contractAddress.toLowerCase()][
+      mapToAPIFormat(currencies, conversionDenomination)
+    ];
+  } catch (e) {
+    console.error(
+      'Could not get contract price from CoinGecko response. Response shape might have changed.',
+      e,
+    );
+    return 0;
+  }
+};
+
+const isContractPriceSuccessResponse = (
+  data: TokenContractPriceResponse,
+): data is TokenContractPriceSuccessResponse => {
+  if (
+    typeof data === 'undefined' || // undefined
+    data == null || // null
```

Comment:
?
```suggestion
    data === null || // null
```

Reply:
Can probably also get rid of the comments, kind of self-explanatory